### PR TITLE
GET /api/articles/:article_id

### PR DESCRIPTION
**Pull request by Howling-Techie**
Completed GET /api/articles/:article_id

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -38,4 +38,37 @@ describe("GET /api/topics", () => {
...
   });
+});
+
+describe("GET /api/articles/:article_id", () => {
+  test("return 200 status code", () => {
+    return request(app).get("/api/articles/1").expect(200);
+  });
+  test("return an article object", () => {
+    return request(app).get("/api/articles/2").then(({body}) => {
+      expect(body.article).toContainAllKeys(["article_id", "title", "topic", "author", "body", "created_at", "votes", "article_img_url"]);
```
Comment (B7Bprog) [inline __tests__/app.test.js:49]: Here I recommend the use of .toMatchObject or .objectContaining so the value types can be checked as well at the same time with 'expect.any'.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -38,4 +38,37 @@ describe("GET /api/topics", () => {
...
+    });
+  });
+  test("return 404 if article not found", () => {
+    return request(app).get("/api/articles/100000").expect(404);
+  });
+  test("return 500 if article_id is not an integer", () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js:55]: This should be a different status code, as we can't really test for a 500 internal server error. Think about what status code this should be when the the request is 'bad'.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -38,4 +38,37 @@ describe("GET /api/topics", () => {
...
+        title: "Eight pug gifs that remind me of mitch",
+        topic: "mitch",
+        author: "icellusedkars",
+        body: "some gifs",
+        created_at: new Date(1604394720000).toISOString(),
+        votes: 0,
+        article_img_url:
+          "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+      };
+      expect(body.article).toEqual(result);
```
Comment (B7Bprog) [inline __tests__/app.test.js:71]: We should avoid testing the object with .toEqual(). Please use a more flexible jest matcher here which will allow you to extend the object in the future without failing this test. Could use one of the ones I mentioned in a previous comment.

File: app.js [hunk 4]
```diff
@@ -1,23 +1,26 @@
 const express = require("express");
 const {getTopics} = require("./controllers/topics.controller");
 const {readFile} = require("fs/promises");
+const {getArticle} = require("./controllers/articles.controller");
 
 const app = express();
 app.use(express.json());
```
Comment (B7Bprog) [inline app.js:7]: What is express.json() responsible for? Do we really need it at this point?

File: models/articles.model.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const db = require("../db/connection");
+
+exports.selectArticle = async (article_id) => {
+  if (Number.isNaN(+article_id)) {
+    return Promise.reject({status: 500, msg: "Invalid article_id query"});
```
Comment (B7Bprog) [inline models/articles.model.js:5]: Wrong status code and message here.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Matt, please see my comments and refactor your code, then send us another PR with it.

File: __tests__/app.test.js [hunk 6]
```diff
@@ -38,4 +38,46 @@ describe("GET /api/topics", () => {
...
+});
+
+describe("GET /api/articles/:article_id", () => {
+  test("return 200 status code", () => {
+    return request(app).get("/api/articles/1").expect(200);
+  });
+  test("return an article object", () => {
+    return request(app).get("/api/articles/2").then(({body}) => {
+      expect(body.article).toMatchObject({
+        article_id: expect.any(Number),
```
Comment (pippaamy) [inline __tests__/app.test.js:50]: we can expect this to be article id 2

File: __tests__/app.test.js [hunk 7]
```diff
@@ -38,4 +38,46 @@ describe("GET /api/topics", () => {
...
+        created_at: expect.any(String),
+        votes: expect.any(Number),
+        article_img_url: expect.any(String)
+      });
+    });
+  });
+  test("return 404 if article not found", () => {
+    return request(app).get("/api/articles/100000").expect(404);
+  });
+  test("return 500 if article_id is not an integer", () => {
```
Comment (pippaamy) [inline __tests__/app.test.js:64]: you need to change the description too

**Review by pippaamy — APPROVED** [review]
good job, a couple changes then all good to merge :)

**Pull request merged**